From 385b06b250ca4ac10ffbb1e36cddbc86745350ce Mon Sep 17 00:00:00 2001
From: David Gibson <dgibson@redhat.com>
Date: Mon, 15 Sep 2014 07:12:37 -0400
Subject: [virt] kvm/ppc: Guard doorbell exception with CONFIG_PPC_DOORBELL

Message-id: <1410765214-16377-22-git-send-email-dgibson@redhat.com>
Patchwork-id: 94716
O-Subject: [PATCH 21/78] KVM: PPC: Guard doorbell exception with CONFIG_PPC_DOORBELL
Bugzilla: 1123145 1123133 1123367
RH-Acked-by: Radim Krcmar <rkrcmar@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1123145

Signed-off-by: David Gibson <dgibson@redhat.com>

commit 5f17ce8b954a2ff93f0460a2ae35d56285697d0f
Author: Tiejun Chen <tiejun.chen@windriver.com>
Date:   Mon May 13 10:00:45 2013 +0800

    KVM: PPC: Guard doorbell exception with CONFIG_PPC_DOORBELL

    Availablity of the doorbell_exception function is guarded by
    CONFIG_PPC_DOORBELL. Use the same define to guard our caller
    of it.

    Signed-off-by: Tiejun Chen <tiejun.chen@windriver.com>
    [agraf: improve patch description]
    Signed-off-by: Alexander Graf <agraf@suse.de>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/kvm/booke.c b/arch/powerpc/kvm/booke.c
index 0385656..4e57c4c 100644
--- a/arch/powerpc/kvm/booke.c
+++ b/arch/powerpc/kvm/booke.c
@@ -775,7 +775,7 @@ static void kvmppc_restart_interrupt(struct kvm_vcpu *vcpu,
   kvmppc_fill_pt_regs(&regs);
   timer_interrupt(&regs);
   break;
-#if defined(CONFIG_PPC_FSL_BOOK3E) || defined(CONFIG_PPC_BOOK3E_64)
+#if defined(CONFIG_PPC_DOORBELL)
  case BOOKE_INTERRUPT_DOORBELL:
   kvmppc_fill_pt_regs(&regs);
   doorbell_exception(&regs);
-- 
1.7.1